### PR TITLE
Support unmarshalling an atomic `Notification` for ordered lists.

### PR DESCRIPTION
**Pull request by wenovus**
Involves supporting `DeleteNode` at the container above the list for ordered maps.

### CODE REVIEW

File: util/gnmi.go [hunk 1]
```diff
@@ -42,6 +42,29 @@ func PathMatchesPrefix(path *gpb.Path, prefix []string) bool {
...
+		return a
+	}
+	return b
+}
+
+// PathPartiallyMatchesPrefix reports whether the path partially or wholly
+// matches the prefix.
+//
+// e.g. a/b partially matches a/b/c or a/b, but doesn't match a/c.
+func PathPartiallyMatchesPrefix(path *gpb.Path, prefix []string) bool {
```
Comment (DanG100) [inline util/gnmi.go:56]: nit: somehow include that no keys are compared in the function name

File: util/gnmi.go [hunk 2]
```diff
@@ -42,6 +42,29 @@ func PathMatchesPrefix(path *gpb.Path, prefix []string) bool {
...
+	for len(prefix) != 0 && prefix[len(prefix)-1] == "" {
```
Comment (DanG100) [inline util/gnmi.go:57]: when are there empty strings in path?

**Review by DanG100 — APPROVED** [review]

File: util/gnmi.go (hunk 2 above)
Comment (wenovus) [reply util/gnmi.go:57]: This is just to guard against a trailing slash when splitting. I wouldn't have written this myself -- it is kept in uniformity with the existing `PathMatchesPrefix`

File: util/gnmi.go (hunk 1 above)
Comment (wenovus) [reply util/gnmi.go:56]: Done.

File: util/gnmi.go (hunk 2 above)
Comment (wenovus) [reply util/gnmi.go:57]: This is just to guard against a trailing slash when splitting. I wouldn't have written this myself -- it is kept in uniformity with the existing `PathMatchesPrefix`

**Pull request merged**